Lib/textwrap.py

```diff
@@ -348,6 +381,10 @@ def wrap(self, text):
         and all other whitespace characters (including newline) are
         converted to space.
         """
+        if self.width <= 0:
+            raise ValueError("invalid width %r (must be > 0)" % self.width)
+        elif self.width == 1 and _width(text) > len(text):
+            raise ValueError("invalid width 1 (must be > 1 when CJK chars)")
```

Comment:
I have done a more complex solution:
```
elif self.width == 1 and (sum(self._width(chunk) for chunk in chunks) >
                              sum(len(chunk) for chunk in chunks)):
```
It throws the exception earlier, but it's probably not absolutely necessary.